modules/api/src/main/java/org/apache/ignite/table/criteria/Criteria.java

```diff
@@ -17,9 +17,226 @@
+     * </pre>
+     *
+     * @param expressions Expressions.
+     * @return The created {@code or} expressions instance.
+     */
+    static Expression or(Expression... expressions) {
+        return new Expression(Operator.OR, expressions);
+    }
+
+    /**
+     * Creates a condition that test the examined object is equal to the specified {@code value}.
+     *
+     * <p>For example:
+     * <pre>
+     *     columnValue(&quot;category&quot;, equalTo(&quot;toys&quot;))
+     * </pre>
+     *
+     * @param <T> Value type.
+     * @param value Target value.
+     */
+    static <T> Condition equalTo(Comparable<T> value) {
+        return new Condition(Operator.EQ, new Parameter<>(value));
+    }
+
+    /**
+     * Creates a condition that test the examined object is equal to the specified {@code value}.
+     *
+     * <p>For example:
+     * <pre>
+     *     columnValue(&quot;category&quot;, equalTo(&quot;toys&quot;))
+     * </pre>
+     *
+     * @param value Target value.
+     */
+    static Condition equalTo(byte[] value) {
+        return new Condition(Operator.EQ, new Parameter<>(value));
+    }
+
+    /**
+     * Creates a condition that test the examined object is greater than the specified {@code value}.
+     *
+     * <p>For example:
+     * <pre>
+     *     columnValue("age", greaterThan(35))
+     * </pre>
+     *
+     * @param <T> Value type.
+     * @param value Target value.
+     */
+    static <T> Condition greaterThan(Comparable<T> value) {
+        return new Condition(Operator.GT, new Parameter<>(value));
+    }
+
+    /**
+     * Creates a condition that test the examined object is greater than or equal than the specified {@code value}.
+     *
+     * <p>For example:
+     * <pre>
+     *     columnValue("age", greaterThanOrEqualTo(35))
+     * </pre>
+     *
+     * @param <T> Value type.
+     * @param value Target value.
+     */
+    static <T> Condition greaterThanOrEqualTo(Comparable<T> value) {
+        return new Condition(Operator.GOE, new Parameter<>(value));
+    }
+
+    /**
+     * Creates a condition that test the examined object is less than the specified {@code value}.
+     *
+     * <p>For example:
+     * <pre>
+     *     columnValue("age", lessThan(35))
+     * </pre>
+     *
+     * @param <T> Value type.
+     * @param value Target value.
+     */
+    static <T> Condition lessThan(Comparable<T> value) {
+        return new Condition(Operator.LT, new Parameter<>(value));
+    }
+
+    /**
+     * Creates a condition that test the examined object is less than or equal than the specified {@code value}.
+     *
+     * <p>For example:
+     * <pre>
+     *     columnValue("age", lessThanOrEqualTo(35))
+     * </pre>
+     *
+     * @param <T> Value type.
+     * @param value Target value.
+     */
+    static <T> Condition lessThanOrEqualTo(Comparable<T> value) {
```

Comment:
we should have tests for all methods for all types

Reply:
Added